genesyscloud/workforcemanagement_businessunits/genesyscloud_workforcemanagement_businessunits_proxy.go

```diff
@@ -0,0 +1,143 @@
+	return p.createWorkforcemanagementBusinessunitsAttr(ctx, p, createBuRequest)
+}
+
+// getWorkforcemanagementBusinessunits retrieves all Genesys Cloud workforcemanagement businessunits
+func (p *workforcemanagementBusinessunitsProxy) getAllWorkforcemanagementBusinessunits(ctx context.Context) (*[]platformclientv2.Businessunitlistitem, *platformclientv2.APIResponse, error) {
+	return p.getAllWorkforcemanagementBusinessunitsAttr(ctx, p)
+}
+
+// getWorkforcemanagementBusinessunitsIdByName returns a single Genesys Cloud workforcemanagement businessunits by a name
+func (p *workforcemanagementBusinessunitsProxy) getWorkforcemanagementBusinessunitsIdByName(ctx context.Context, name string) (string, *platformclientv2.APIResponse, bool, error) {
+	return p.getWorkforcemanagementBusinessunitsIdByNameAttr(ctx, p, name)
+}
+
+// getWorkforcemanagementBusinessunitsById returns a single Genesys Cloud workforcemanagement businessunits by ID
+func (p *workforcemanagementBusinessunitsProxy) getWorkforcemanagementBusinessunitsById(ctx context.Context, id string) (*platformclientv2.Businessunitresponse, *platformclientv2.APIResponse, error) {
+	return p.getWorkforcemanagementBusinessunitsByIdAttr(ctx, p, id)
+}
+
+// updateWorkforcemanagementBusinessunits updates a Genesys Cloud workforcemanagement businessunits
+func (p *workforcemanagementBusinessunitsProxy) updateWorkforcemanagementBusinessunits(ctx context.Context, id string, updateRequest *platformclientv2.Updatebusinessunitrequest) (*platformclientv2.Businessunitresponse, *platformclientv2.APIResponse, error) {
+	return p.updateWorkforcemanagementBusinessunitsAttr(ctx, p, id, updateRequest)
+}
+
+// deleteWorkforcemanagementBusinessunits deletes a Genesys Cloud workforcemanagement businessunits by Id
+func (p *workforcemanagementBusinessunitsProxy) deleteWorkforcemanagementBusinessunits(ctx context.Context, id string) (*platformclientv2.APIResponse, error) {
+	return p.deleteWorkforcemanagementBusinessunitsAttr(ctx, p, id)
+}
+
+// createWorkforcemanagementBusinessunitsFn is an implementation function for creating a Genesys Cloud workforcemanagement businessunits
+func createWorkforcemanagementBusinessunitsFn(ctx context.Context, p *workforcemanagementBusinessunitsProxy, workforcemanagementBusinessunits *platformclientv2.Createbusinessunitrequest) (*platformclientv2.Businessunitresponse, *platformclientv2.APIResponse, error) {
+	return p.workforceManagementApi.PostWorkforcemanagementBusinessunits(*workforcemanagementBusinessunits, false)
+}
+
+// getAllWorkforcemanagementBusinessunitsFn is the implementation for retrieving all workforcemanagement businessunits in Genesys Cloud
+func getAllWorkforcemanagementBusinessunitsFn(ctx context.Context, p *workforcemanagementBusinessunitsProxy) (*[]platformclientv2.Businessunitlistitem, *platformclientv2.APIResponse, error) {
+	businessUnitResponses, resp, err := p.workforceManagementApi.GetWorkforcemanagementBusinessunits("", "")
```

Comment:
why the empty strings? Again, not too familiar with go, but is there the concept of optional parameters or default parameters? And is null not a first class thing kotlin (and why you are using empty string)

Reply:
The two parameters being passed are `feature string, divisionId string`. as mentioned previously, strings can't be null in Go. Further note: This is a call into the actual generated Go sdk so can't edit the method to take pointers